src/emu.h

```diff
@@ -0,0 +1,111 @@
+#include <iostream>
+
+using byte_t = uint8_t;
+using word_t = uint16_t;
+template<typename T> bool IsPtrAligned(word_t ptr);
+template<> bool IsPtrAligned<byte_t>(word_t ptr) { return true; }
+//template<> bool IsPtrAlgined<word_t>(word_t ptr) { return !(ptr % sizeof(word_t)); }
+
+struct Emu {
+	enum struct TrapVec : byte_t {
+		USER_DEV	= 0000,
```

Comment:
Please add comments explaining when are these traps invoked, or at least those that will presumably be useful to us. At the very least please provide sources of these constants, e.g. here
`// Refer to page 229, PDP1145`

Reply:
We will not copy the full handbook to our code. The trap vector is located in the reserved memory, refer to PDP-11/45 Handbook, Appendix C.

Reply:
I should add comments to track implementation status of each trap type.